.github/workflows/manual-deploy-dev-staging.yml

```
@@ -18,11 +18,96 @@ on:
 env:
   DEVOPS_CHANNEL_ID: C37M86Y8G #devops-deploys
   VETS_WEBSITE_CHANNEL_ID: C02V265VCGH # status-vets-website
+  BUILD_ENV_DEV: vagovdev
+  BUILD_ENV_STAGING: vagovstaging
 
 jobs:
+  build:
+    name: Build
+    runs-on: self-hosted
+
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        uses: ./.github/workflows/install
+        with:
+          key: ${{ hashFiles('yarn.lock') }}
+          yarn_cache_folder: .cache/yarn
+          path: |
+            .cache/yarn
+            node_modules
+
+      - name: Build dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: yarn build --verbose --buildtype=vagovdev
+        timeout-minutes: 30  
+        
+      - name: Generate build details for dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: |
+          cat > build/${{ env.BUILD_ENV_DEV }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV_DEV }}
+          NODE_ENV=dev
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          CHANGE_TARGET=null
+          RUN_ID=${{ github.run_id }}
+          RUN_NUMBER=${{ github.run_number }}
+          REF=${{ github.sha }}
+          BUILDTIME=$(date +%s)
+          EOF
+
+      - name: Compress and archive build for dev 
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: tar -C build/${{ env.BUILD_ENV_DEV }} -cjf ${{ env.BUILD_ENV_DEV }}.tar.bz2 .
+
+      - name: Upload build artifact for dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.BUILD_ENV_DEV }}.tar.bz2
+          path: ${{ env.BUILD_ENV_DEV }}.tar.bz2
+          retention-days: 1   
+        
+      - name: Build staging
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        run: yarn build --verbose --buildtype=vagovstaging
+        timeout-minutes: 30 
+
+      - name: Generate build details for staging 
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        run: |
+          cat > build/${{ env.BUILD_ENV_STAGING }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV_STAGING }}
+          NODE_ENV=staging
```

```suggestion
          NODE_ENV=production
```
We have the environment variable set to `production` in all builds 
https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/continuous-integration.yml#L66

---

Thanks for the review, looking.